### PR TITLE
Improved pretty printing in boot

### PR DESCRIPTION
**Pull request by dlunde**
I've added additional pretty printing functionality to boot, mostly to make it easier for myself to learn how the mlang transformations work. Maybe others can benefit from this as well.

### CODE REVIEW

File: src/boot/boot.ml [hunk 1]
```diff
@@ -211,27 +219,49 @@ let testprog lst =
 
 (* Run program *)
 let runprog name lst =
+    (* TODO prog_argv is never used anywhere *)
```
Comment (elegios) [inline src/boot/boot.ml:222]: Looking at the code in `mexpr.ml` it appears to grab `argv` separately, maybe this is left from some earlier thought on how it should be done?

(not particularly relevant to this PR, but maybe worth considering later)

File: src/boot/boot.ml [hunk 2]
```diff
@@ -211,27 +219,49 @@ let testprog lst =
...
 
 (* Main function. Checks arguments and reads file names *)
 let main =
 
-  (* Check command  *)
-  (match Array.to_list Sys.argv |> List.tl with
+  (* A list of command line arguments *)
+  let speclist = [
+
+    (* First character in description string must be a space for alignment! *)
```
Comment (elegios) [inline src/boot/boot.ml:236]: Is this something that `Arg` dictates?

File: src/boot/pprint.ml [hunk 3]
```diff
@@ -1,244 +1,492 @@
-
-(*
-   Miking is licensed under the MIT license.
-   Copyright (C) David Broman. See file LICENSE.txt
-*)
+(** Pretty-printing for data types in Miking.
+ *
+ *  The main functions that should be used outside of this module is
```
Comment (elegios) [inline src/boot/pprint.ml:3]: Maybe we should consider making a `.mli`? Maybe overkill for `boot`?

File: src/boot/pprint.ml [hunk 4]
```diff
@@ -1,244 +1,492 @@
...
+    print_const c
+
+(** Converting environments to strings.
+ *  TODO Messy with optional arguments passing. Alternatives? *)
+let string_of_env ?debruijn ?indent ?max_indent ?margin ?max_boxes ?prefix e =
+  str_formatter_print ?debruijn ?indent ?max_indent ?margin ?max_boxes ?prefix
+    print_env e
+
+(** Old printing function for terms.
+ *  @deprecated Use string_of_tm instead. *)
```
Comment (elegios) [inline src/boot/pprint.ml]: Question for meeting: do we want to deprecate and then remove, or just remove and make necessary changes?

**Review by elegios — APPROVED** [review]

File: src/boot/boot.ml (hunk 2 above)
Comment (dlunde) [reply src/boot/boot.ml:236]: Yes!

File: src/boot/pprint.ml (hunk 3 above)
Comment (dlunde) [reply src/boot/pprint.ml:3]: I thought about this as well

**Comment by dlunde** [comment]
The comments from the previous meeting has now been addressed:
- Removed deprecated functions (this also resolved camel vs snake case)
- Return ustrings instead of strings

**Pull request merged**